sdk/identity/Azure.Identity/src/IdentityClient.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.Pipeline.Policies;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    internal class IdentityClient
+    {
+        private readonly IdentityClientOptions _options;
+        private readonly HttpPipeline _pipeline;
+        
+
+        public IdentityClient(IdentityClientOptions options = null)
+        {
+            _options = options ?? new IdentityClientOptions();
+
+            _pipeline = HttpPipeline.Build(_options,
+                    _options.ResponseClassifier,
+                    _options.RetryPolicy,
+                    ClientRequestIdPolicy.Singleton,
+                    _options.LoggingPolicy,
+                    BufferResponsePolicy.Singleton);
+        }
+
+        public async Task<AuthenticationResponse> AuthenticateAsync(string tenantId, string clientId, string clientSecret, IEnumerable<string> scopes, CancellationToken cancellationToken = default)
+        {
+            Uri authUri = new Uri(_options.AuthorityHost + tenantId + "/oauth2/token");
+
+            using (var request = _pipeline.CreateRequest())
+            {
+                request.SetRequestLine(HttpPipelineMethod.Put, authUri);
+
+                ReadOnlyMemory<byte> content = Serialize(
+                    ( "response_type", "token" ), 
+                    ( "grant_type", "client_credentials" ),
+                    ( "client_id", clientId ),
+                    ( "client_secret", clientSecret ),
+                    ( "scopes", string.Join(" ", scopes) ));
+
+                request.Content = HttpPipelineRequestContent.Create(content);
```

Comment:
@parkym, should we have an implementation of IBufferWriter that creates HttpPipelineContent? It can do much better job sizing/resizing the array and simplify code.

Reply:
Yes, after package update PR goes in I'll add something.